src/rc/start-stop-daemon.c

```diff
@@ -1005,7 +1024,15 @@ int main(int argc, char **argv)
 
 		for (i = getdtablesize() - 1; i >= 3; --i)
 			close(i);
-
+#ifdef HAVE_SCHED
+		if (scheduler != -1){
+			sched = malloc(sizeof(sched));
```

Comment:
why does this need to be allocated on the heap ?  just declare it as a local stack variable.